File: board_dsl/src/main/java/com/dooho/board/api/board/search/SearchEntity.java
package com.dooho.board.api.board.search;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity(name="Search")
@Table(name="Search")
public class SearchEntity {
    @Id
    private String popularTerm;
    private Integer popularSearchCount;

    public SearchEntity(SearchDto dto) {
        this.popularTerm = dto.getPopularTerm();
        this.popularSearchCount = dto.getPopularSearchCount();
    }

    public void setPopularSearchCount(Integer popularSearchCount) {
        this.popularSearchCount = popularSearchCount;
    }
}

File: inflearn_basic_DB/board_Kotlin/src/main/java/com/dooho/board/api/board/search/SearchEntity.kt
package com.dooho.board.api.board.search

import com.dooho.board.api.board.search.SearchDto
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity(name = "Search")
@Table(name = "Search")
data class SearchEntity(
    @Id
    var popularTerm: String? = null,
    var popularSearchCount: Int? = null
) {
    constructor(dto: SearchDto?) : this(
        popularTerm = dto?.popularTerm,
        popularSearchCount = dto?.popularSearchCount
    )
}